pkg/core/native_contract_test.go

```diff
@@ -139,3 +145,59 @@ func TestNativeContract_Invoke(t *testing.T) {
 		require.Fail(t, "onPersist wasn't called")
 	}
 }
+
+func TestNativeContract_InvokeInternal(t *testing.T) {
+	chain := newTestChain(t)
+	defer chain.Close()
+
+	tn := newTestNative()
+	chain.registerNative(tn)
+
+	err := chain.dao.PutContractState(&state.Contract{
+		Script:   tn.meta.Script,
+		Manifest: tn.meta.Manifest,
+	})
+	require.NoError(t, err)
+
+	t.Run("fail, direct invocation", func(t *testing.T) {
```

Comment:
Why do these 2 tests have different structure? Can we use `native.Call` directly here?

Reply:
We can use `native.Call` directly in both test cases. However, the first `fail, direct invocation` testcase is more complicated and has wider coverage than the second `success` one. I'd prefer to make the second case looks like the first one, but it's not possible because for success direct invocation it's required for calling script hash and contract script hash to be equal. So I decided to keep such structure for at least the first testcase. Should I replace it by `native.Call` for the first testcase?